transport/server.cc

```diff
@@ -866,16 +871,20 @@ cql_server::connection::process_on_shard(::shared_ptr<messages::result_message::
             return process_fn(client_state, server._query_processor, in, stream, _version, _cql_serialization_format,
                     /* FIXME */empty_service_permit(), std::move(trace_state), false, std::move(cached_vals)).then([] (auto msg) {
                 // result here has to be foreign ptr
-                return std::get<foreign_ptr<std::unique_ptr<cql_server::response>>>(std::move(msg));
+                return std::get<coordinator_result<foreign_ptr<std::unique_ptr<cql_server::response>>>>(std::move(msg)).value();
             });
         });
     });
 }
 
 using process_fn_return_type = std::variant<
-    foreign_ptr<std::unique_ptr<cql_server::response>>,
+    coordinator_result<foreign_ptr<std::unique_ptr<cql_server::response>>>,
     ::shared_ptr<messages::result_message::bounce_to_shard>>;
 
+static inline coordinator_result<foreign_ptr<std::unique_ptr<cql_server::response>>> convert_error_message_to_coordinator_result(messages::result_message* msg) {
+    return bo::failure(std::move(*dynamic_cast<messages::result_message::exception*>(msg)).get_exception());
```

Comment:
Can't we create a coordinator_result<> directly from the exception without using bo::failure?
If we could we could avoid using that ugly namespace in this source file.

By the way, I see you ended up using dynamic_cast<> as I suggested. So that's yet another reason not to use the std::variant<> and just use a common base class.

Reply:
The dynamic_cast here is not used to cast between exception types. It's used to cast a result_message into result_message::exception.

As for the `bo::failure`, you are right that we can create a coordinator_result<> directly from the exception. I'll remove `bo::failure`.